- added playlist creation

dencharski opened the pull request.

dencharski committed:
- added playlist creation

IliaAnK commented on app/src/main/java/com/example/playlistmaker/mediateka/domain/impl/PlayListInteractorImpl.kt:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    override suspend fun insertTrackDtoAppInPlayList(trackDtoApp: TrackDtoApp) {
+        addTrackInPlayListRepository.addTrackInPlayList(trackDbConvertor.map(trackDtoApp))
```
Обязательно:
Маппинг из `domain` в `data` и обратно должен происходить на стороне `data` слоя.
`domain` не должен ничего знать про сторонние зависимости, в том числе про `DTO` и т.п.

Аналогичная рекомендация и для других интеракторов!

IliaAnK commented on app/src/main/java/com/example/playlistmaker/audio_player/ui/AudioPlayerViewModel.kt:
```diff
@@ -28,26 +31,69 @@ class AudioPlayerViewModel(
+    }
+
+    fun checkTrackInPlaylist(playList: PlayList) {
+        if (playList.listOfTrackIds.contains(_currentTrackId)) {
```
Обязательно:
Вот эту логику нужно перенести в репозиторий.
Суть в том, что откуда бы не был добавлен трек нужно проверить его на наличие в базе данных, если есть вернуть значение, что есть, если нет то записать трек в базу.
Так получается, что достаточно будет вызвать один метод репозитория и не писать эту логику снова.

А если логика поменяется, то достаточно будет изменить её только в одном месте.

dencharski committed:
- added playlist creation.
fixed playlist image,
fixed add track in playlist,
fixed release date,
fixed logic in the data layer.
